sdk/samples/iot/provisioning/src/paho_iot_provisioning_example.c

```diff
@@ -316,168 +310,161 @@ static int register_device()
+  // Continue to parse incoming responses from the Device Provisioning Service until
   // the device has been successfully provisioned or an error occurs
   do
   {
-    // Receive message.
-    // If MQTTClient_receive is successful, topic and message memory must later be freed
-    if ((rc
-         = MQTTClient_receive(mqtt_client, &topic, &topic_len, &message, TIMEOUT_MQTT_RECEIVE_MS))
-        != MQTTCLIENT_SUCCESS)
+    if (((rc
+          = MQTTClient_receive(mqtt_client, &topic, &topic_len, &message, TIMEOUT_MQTT_RECEIVE_MS))
+         != MQTTCLIENT_SUCCESS)
+        && (MQTTCLIENT_TOPICNAME_TRUNCATED != rc))
     {
-      (void)printf("Failed to receive message: MQTTClient return code %d .\n", rc);
-      return rc;
+      LOG_ERROR("Failed to receive message: MQTTClient return code %d .\n", rc);
+      exit(rc);
     }
-    (void)printf("Received a message from service.\n");
-
-    // Parse operation message
-    if (az_failed(
-            rc = parse_operation_message(topic, topic_len, message, &response, &operation_status)))
+    else if (!message)
     {
-      (void)printf("Failed to parse operation message: az_result return code 0x%04x .\n", rc);
-      MQTTClient_freeMessage(&message);
-      MQTTClient_free(topic);
-      return rc;
+      LOG_ERROR("Timeout expired: MQTTClient return code %d .\n", rc);
+      exit(rc);
+    }
+    else if (MQTTCLIENT_TOPICNAME_TRUNCATED == rc)
+    {
+      topic_len = (int)strlen(topic);
     }
-    (void)printf("Parsed operation message.\n");
+    LOG_SUCCESS("Client received a message from the provisioning service.\n");
+
+    parse_operation_message(topic, topic_len, message, &response, &operation_status);
+    LOG_SUCCESS("Client parsed operation message.\n");
 
     // If operation is not complete, send query and loop to receive operation message
     is_operation_complete = az_iot_provisioning_client_operation_complete(operation_status);
     if (!is_operation_complete)
     {
+      (void)printf("\t Operation incomplete.\n");
       MQTTClient_freeMessage(&message);
       MQTTClient_free(topic);
 
-      if ((rc = send_operation_query_message(&response)) != MQTTCLIENT_SUCCESS)
-      {
-        return rc;
-      }
+      send_operation_query_message(&response);
+      LOG_SUCCESS("Client sent operation query message.\n");
     }
   } while (!is_operation_complete);
 
-  // Operation is complete
-  // Successful assignment - print out the assigned hostname and device id
+  // Operation is complete: Successful assignment
   if (operation_status == AZ_IOT_PROVISIONING_STATUS_ASSIGNED)
   {
-    (void)printf("SUCCESS - Device provisioned:\n");
-    print_az_span("\tHub Hostname: ", response.registration_result.assigned_hub_hostname);
-    print_az_span("\tDevice Id: ", response.registration_result.device_id);
+    LOG_SUCCESS("Client provisioned:\n");
+    print_az_span("\t Hub Hostname: ", response.registration_result.assigned_hub_hostname);
```

Comment:
I see a potential issue here with text not aligning because people don't know the ultimate rendering of the text. Meaning here, you probably know to add a space to get it to align. Can the `print_az_span` underneath call `LOG_SUCCESS` which manages the alignment?

Reply:
With maybe a parameter marking it as a success or failure?